File: fixtures/scale/objc/large/Families/Crc32/Crc32Descriptor.m
#import "Crc32Descriptor.h"
#import "Crc32Cipher.h"
#import "Crc32Vectors.h"

@implementation Crc32Descriptor

- (NSString *)family {
    return @"crc32";
}

- (NSString *)suite {
    return @"hash";
}

- (id<CLCipher>)cipher {
    return [[Crc32Cipher alloc] init];
}

- (NSArray *)vectors {
    return [Crc32Vectors all];
}

@end

File: fixtures/scale/php/large/src/Families/Crc32/Crc32Descriptor.php
<?php

namespace CipherLab\Families\Crc32;

use CipherLab\Abstractions\Cipher;
use CipherLab\Abstractions\FamilyDescriptor;

// Registered once in FamilyCatalog; ties together Crc32Cipher and
// Crc32Vectors under the "hash" suite.
class Crc32Descriptor implements FamilyDescriptor
{
    public function family(): string
    {
        return 'crc32';
    }

    public function suite(): string
    {
        return 'hash';
    }

    public function cipher(): Cipher
    {
        return new Crc32Cipher();
    }

    public function vectors(): array
    {
        return Crc32Vectors::all();
    }
}
